cmd/generate/dockerfile-templates/FuncUtilDockerfile.template

```diff
@@ -4,6 +4,7 @@ ARG GO_RUNTIME=registry.access.redhat.com/ubi8/ubi-minimal
 
 FROM $GO_BUILDER as builder
 
+WORKDIR /go/src/{{.main}}
```

Comment:
Couldn't we have just `WORKDIR /workspace` for sake of simplicity? I think that for long time project does not have to be in `$GOPATH/src`. 

And even if we wanted to use the old way then the path should **not** be `/go/src/{{.main}}` (evaluated to `/go/src/cmd/func-util`).
It should be more something like `/go/src/{{.module}}` (evaluated to something like `/go/src/github.com/knative/func`).

Reply:
`/go/src/{{.main}}` just doesn't seems right even thought it might work.

Reply:
See [GOPATH development mode](https://go.dev/wiki/GOPATH#gopath-development-mode) vs [Deprecating and removing GOPATH development mode](https://go.dev/wiki/GOPATH#deprecating-and-removing-gopath-development-mode).

Reply:
It can be pretty much anything. For the issue with multi-arch builds, the only problem was staying on root `/`. However, it's only workdir. There are no other build consequence AFAIK. The convenience was that we had `{{.main}}` already present in the template. It can be whatever, but to it's a build impl detail that doesn't matter unless it's breaking anything.

Reply:
updated